integrations/mistral/tests/test_ocr_document_converter.py

```diff
@@ -0,0 +1,849 @@
+        assert converter.include_image_base64 is False
+        assert converter.pages is None
+        assert converter.image_limit is None
+        assert converter.image_min_size is None
+        assert converter.cleanup_uploaded_files is True
+
+    def test_from_dict_with_custom_parameters(self, monkeypatch):
+        monkeypatch.setenv("MISTRAL_API_KEY", "test-api-key")
+        converter_dict = {
+            "type": (
+                "haystack_integrations.components.converters.mistral."
+                "ocr_document_converter.MistralOCRDocumentConverter"
+            ),
+            "init_parameters": {
+                "api_key": {
+                    "env_vars": ["MISTRAL_API_KEY"],
+                    "strict": True,
+                    "type": "env_var",
+                },
+                "model": "mistral-ocr-custom",
+                "include_image_base64": True,
+                "pages": [0, 1, 2],
+                "image_limit": 10,
+                "image_min_size": 100,
+                "cleanup_uploaded_files": False,
+            },
+        }
+
+        converter = MistralOCRDocumentConverter.from_dict(converter_dict)
+
+        assert converter.model == "mistral-ocr-custom"
+        assert converter.include_image_base64 is True
+        assert converter.pages == [0, 1, 2]
+        assert converter.image_limit == 10
+        assert converter.image_min_size == 100
+        assert converter.cleanup_uploaded_files is False
+
+    @pytest.fixture
+    def mock_ocr_response(self):
+        """Create a mock OCR response"""
+        mock_page = MagicMock()
+        mock_page.markdown = "# Sample Document\n\nThis is page 1."
+        mock_page.images = []
+
+        mock_response = MagicMock()
+        mock_response.pages = [mock_page]
+        mock_response.document_annotation = None
+        mock_response.model_dump.return_value = {
+            "pages": [
+                {"markdown": "# Sample Document\n\nThis is page 1.", "images": []}
+            ],
+            "document_annotation": None,
+        }
+        return mock_response
+
+    @pytest.fixture
+    def mock_ocr_response_with_multiple_pages(self):
+        """Create a mock OCR response with multiple pages"""
+        mock_page1 = MagicMock()
+        mock_page1.markdown = "# Page 1"
+        mock_page1.images = []
+
+        mock_page2 = MagicMock()
+        mock_page2.markdown = "# Page 2"
+        mock_page2.images = []
+
+        mock_response = MagicMock()
+        mock_response.pages = [mock_page1, mock_page2]
+        mock_response.document_annotation = None
+        mock_response.model_dump.return_value = {
+            "pages": [
+                {"markdown": "# Page 1", "images": []},
+                {"markdown": "# Page 2", "images": []},
+            ],
+            "document_annotation": None,
+        }
+        return mock_response
+
+    def test_run_with_document_url_chunk(self, mock_ocr_response):
```

Comment:
the following tests are almost identical, so we can use `@pytest.mark.parametrize` to avoid duplication